server/bin/pbench-server-prep-shim-002.py

```diff
@@ -0,0 +1,246 @@
+    if not cfg_name:
+        print(f"{_NAME_}: ERROR: No config file specified; set CONFIG")
+        return 2
+
+    try:
+        config = PbenchConfig(cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e} (config file {cfg_name})", file=sys.stderr)
+        return 1
+
+    if not os.path.exists(config.ARCHIVE):
+        print(f"{_NAME_}: Bad ARCHIVE={config.ARCHIVE}", file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    qdir = config.get("pbench-server", "pbench-quarantine-dir")
+    if len(qdir) == 0:
+        logger.error("Failed: getconf.py pbench-quarantine-dir pbench-server")
+        return 2
+
+    if not os.path.isdir(qdir):
+        logger.error("Failed: {} does not exist, or is not a directory", qdir)
+        return 2
+
+    receive_dir_prefix = config.get("pbench-server", "pbench-receive-dir-prefix")
+    if len(receive_dir_prefix) == 0:
+        logger.error("Failed: getconf.py pbench-receive-dir-prefix pbench-server")
+        return 2
+
+    receive_dir = f"{receive_dir_prefix}-002"
+    if not os.path.isdir(receive_dir):
+        logger.error("Failed: {} does not exist, or is not a directory", receive_dir)
+        return 2
+
+    qdir_md5 = os.path.join(qdir, "md5-002")
+    if not os.path.exists(qdir_md5):
+        try:
+            os.makedirs(qdir_md5)
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.exception(
+                "os.mkdir: Unable to create quarantine destination directory: {}",
+                qdir_md5,
+            )
+            return 1
+
+    duplicates = os.path.join(qdir, "duplicates-002")
+    if not os.path.exists(duplicates):
+        try:
+            os.makedirs(duplicates)
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.exception(
+                "os.mkdir: Unable to create duplicates destination directory: {}",
+                duplicates,
+            )
+            return 1
+
+    # The following directory holds tarballs that are quarantined because
+    # of operational errors on the server. They should be retried after
+    # the problem is fixed: basically, move them back into the reception
+    # area for 002 agents and wait.
+
+    errors = os.path.join(qdir, "errors-002")
+    if not os.path.exists(errors):
+        try:
+            os.makedirs(errors)
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.exception(
+                "os.mkdir: Unable to create errors destination directory: {}", errors
+            )
+            return 1
+
+    list_check = list()
+    receive_dir_check = glob.iglob(os.path.join(receive_dir, "**"), recursive=True)
+    for tar in receive_dir_check:
+        if tar.endswith(".tar.xz.md5"):
```

Comment:
Can we put this pattern right into the `iglob()` call?

Something like:

```
list_check = glob.glob(os.path.join(receive_dir, "**", "*.tar.xz.md5", recursive=True)
```

Which would replace lines 103 thru 107.